samples/net/openthread/coap_utils/coap_utils.c

```diff
@@ -0,0 +1,242 @@
+#include <zephyr.h>
+#include <errno.h>
+#include <logging/log.h>
+#include <net/coap.h>
+#include <net/net_ip.h>
+#include <net/net_mgmt.h>
+#include <net/socket.h>
+#include <net/udp.h>
+#include <sys/byteorder.h>
+#include <sys/printk.h>
+
+#include "coap_utils.h"
+
+LOG_MODULE_REGISTER(coap_utils, CONFIG_COAP_UTILS_LOG_LEVEL);
+
+#define MAX_COAP_MSG_LEN 256
+#define COAP_VER 1
+#define COAP_TOKEN_LEN 8
+#define COAP_MAX_REPLIES 1
+#define COAP_POOL_SLEEP 500
+#define COAP_OPEN_SOCKET_SLEEP 200
+#define COAP_RECEIVE_STACK_SIZE 500
+
+static K_THREAD_STACK_DEFINE(receive_stack_area, COAP_RECEIVE_STACK_SIZE);
+static struct k_thread receive_thread_data;
+
+const static int nfds = 1;
+static struct pollfd fds;
+static struct coap_reply replies[COAP_MAX_REPLIES];
+
+static int coap_open_socket(void)
+{
+	int sock;
+
+	while (1) {
+		sock = socket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP);
+		if (sock < 0) {
+			LOG_ERR("Failed to create UDP socket %d", errno);
+			k_sleep(COAP_OPEN_SOCKET_SLEEP);
+			continue;
+		}
+		break;
+	}
+
+	return sock;
+}
+
+static void coap_close_socket(int socket)
+{
+	(void)close(socket);
+}
+
+static void coap_receive(void)
+{
+	static u8_t buf[MAX_COAP_MSG_LEN + 1];
+
+	struct coap_packet response;
+	struct coap_reply *reply = NULL;
+	static struct sockaddr from_addr;
+	socklen_t from_addr_len;
+	int len;
+	int ret;
+
+	while (1) {
+		if (poll(&fds, nfds, K_FOREVER) < 0) {
+			LOG_ERR("Error in poll:%d", errno);
+			errno = 0;
+			k_sleep(COAP_POOL_SLEEP);
+			continue;
+		}
+
+		if (fds.revents & POLLERR) {
+			LOG_ERR("Error in poll.. waiting a moment.");
+			fds.revents = 0;
```

Comment:
I think all the fds.revents = 0; can be replaced with one just before the if (poll(&fds, nfds, K_FOREVER) < 0)